src/client/datascience/notebook/kernelProvider.ts

```diff
@@ -125,12 +144,18 @@ export class VSCodeKernelPickerProvider implements INotebookKernelProvider {
 
         // If no preferred kernel set but we have a language, use that to set preferred instead.
         if (!mapped.find((v) => v.isPreferred)) {
-            const languages = document.cells.map((c) => c.language);
+            const languages = Array.from(new Set<string>(document.cells.map((c) => c.language)).values());
```

Comment:
Arrary.from can just convert the Set object. Shouldn't need the .values() here.

Reply:
Oh wow, didn't know that. thanks